tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/MvcOptionsIntegration.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="MvcOptionsIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if !NETFRAMEWORK
+using System.ComponentModel;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore;
+
+/// <summary>
+/// The ASP.NET Core middleware integration.
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "Microsoft.AspNetCore.Mvc.Core",
+    TypeName = TypeName,
+    MethodName = ".ctor",
+    ReturnTypeName = ClrNames.Void,
+    MinimumVersion = Major2,
+    MaximumVersion = Major7,
+    IntegrationName = nameof(IntegrationId.AspNetCore),
+    InstrumentationCategory = InstrumentationCategory.AppSec)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class MvcOptionsIntegration
+{
+    private const string Major2 = "2";
+    private const string Major7 = "7";
+    private const string TypeName = "Microsoft.AspNetCore.Mvc.MvcOptions";
+
+    internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, System.Exception exception, in CallTargetState state)
+        where TTarget : IMvcOptions
+    {
+        // dont test appsec enabled here, as there is no way to add a filter later on
+        instance.Filters.Add(new ActionResponseFilter());
```

Comment:
There's nothing stopping customers removing this filter (accidentally) here. You may (potentially) want to _consider_ instrumenting the place where these options are actually _used_.

Also, you may need to consider [the order of execution of global vs local filters here](https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-7.0#default-order-of-execution). By adding a global filter, you're ensuring your filter runs the executed step _last_, but that means there's more chance your filter will be short-circuited...